ir/state.cpp

```diff
@@ -149,6 +151,16 @@ expr State::strip_undef_and_add_ub(const Value &val, const expr &e) {
            is_undef_cond(not_undef, var);
   };
 
+  auto is_if_undef_or_add = [&](const expr &e, expr &var, expr &not_undef) {
```

Comment:
this is unrelated to the main patch. Can you split this and benchmark the those patches independently pls? Otherwise we have no idea what's causing what.